src/software/ai/hl/stp/play/kickoff_enemy_play_test.cpp

```diff
@@ -28,18 +31,62 @@ TEST_F(KickoffEnemyPlayTest, test_kickoff_enemy_play)
     setRefereeCommand(RefereeCommand::NORMAL_START, RefereeCommand::PREPARE_KICKOFF_THEM);
 
     std::vector<ValidationFunction> terminating_validation_functions = {
-        // This will keep the test running for 9.5 seconds to give everything enough
-        // time to settle into position and be observed with the FullSystemGUI
-        // TODO: Implement proper validation
-        // https://github.com/UBC-Thunderbots/Software/issues/1397
         [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
-            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(9.5))
+            auto robotsShadowingEnemy = [](std::shared_ptr<World> world_ptr) -> bool {
+                // TODO: Fix bug with robot three not shadowing the enemy kicker in
+                // kickoff_enemy_play
+                // https://github.com/UBC-Thunderbots/Software/issues/1945
+
+                Rectangle robotOneShadowingRect(Point(0, 2.2), Point(-0.4, 1.8));
+                Rectangle robotFiveShadowingRect(Point(0, -2.2), Point(-0.4, -1.8));
+
+                Point robotOnePos =
+                    world_ptr->friendlyTeam().getRobotById(1).value().position();
+                Point robotFivePos =
+                    world_ptr->friendlyTeam().getRobotById(5).value().position();
+
+                return contains(robotOneShadowingRect, robotOnePos) &&
+                       contains(robotFiveShadowingRect, robotFivePos);
+            };
+
+            auto robotsDefendingPosts = [](std::shared_ptr<World> world_ptr) -> bool {
```

Comment:
@akhilveeraghanta thoughts on checking this since we might be replacing this with creasedefenders?

Reply:
Yeah we probably don't need to check for the exact location of the goal post defenders, easier to just use crease defenders here. 

maybe we can just generalize it like @itsarune suggested below